packages/fuels-contract/src/contract.rs

```diff
@@ -514,14 +532,43 @@ where
 
     /// Call a contract's method on the node, in a state-modifying manner.
     pub async fn call(self) -> Result<CallResponse<D>, Error> {
-        Self::call_or_simulate(self, false).await
+        Self::call_or_simulate(&self, false).await
     }
 
     /// Call a contract's method on the node, in a simulated manner, meaning the state of the
     /// blockchain is *not* modified but simulated.
     /// It is the same as the `call` method because the API is more user-friendly this way.
     pub async fn simulate(self) -> Result<CallResponse<D>, Error> {
-        Self::call_or_simulate(self, true).await
+        Self::call_or_simulate(&self, true).await
+    }
+
+    /// Simulates the call and attempts to resolve missing tx dependencies.
+    /// Forwards the received error if it cannot be fixed.
+    pub async fn estimate_tx_dependencies(
+        mut self,
+        max_attempts: Option<u64>,
```

Comment:
What is the logical reason for allowing the user to provide a maximum attempt? I feel like this argument would just force the user to pass in `None` 99.99% of the time. What do you think?

Reply:
The reason is to provide granular control of a latency-sensitive part of their code. Yes, a big part of the users won't care about this control; they can just pass in `None`. The users who care will be triggered if they cannot control the number of _network trips_ of a function call that could be in a latency-sensitive part of their code.

Reply:
Ah yes I overlooked the network part, very touchy subject indeed haha.